### PR TITLE
Chopsticks & E2E Tests

### PR DESCRIPTION
**Pull request by gluneau**
Benefits of using chopsticks and examples of CLI

### CODE REVIEW

**Review by andabak — DISMISSED** [review]
General comments:

- You should cross link to Chopsticks documents for more reference (I did not notice this in the text)
- also, based on Learning Session - commenting here but relates to he Astar config file on Chopstick repo - there should be some comments / explanation of  Astar config and why it is  set up the way it is (DOTs on Alice, USDT on Alice etc) (add as comments in that YAML file for sure)

File: docs/build/environment/chopsticks.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+2. Enhanced performance: Testing transactions and upgrades on a forked version of the blockchain enables developers to analyze the performance implications of their changes, such as transaction throughput and latency. This can lead to optimizations that enhance the overall performance of the main blockchain network.
+
+3. Reduced downtime: Forking a live blockchain for testing purposes can help minimize network downtime during upgrades or other maintenance activities, as it allows developers to thoroughly test and validate changes before implementation on the main network.
+
+4. Increased user confidence: Users can feel more confident in the stability and reliability of the blockchain network when they know that proposed changes have been rigorously tested in an environment that closely mirrors the live network.
+
+5. Facilitated innovation: Forking a live blockchain provides a space for experimentation, allowing developers to test new features, protocols, and consensus algorithms without disrupting the main network. This fosters innovation and can lead to the development of novel applications and use cases for blockchain technology.
```
Comment (andabak) [inline docs/build/environment/chopsticks.md:21]: Also, I would not overpromise all of these statements - Lets leave this to Chopsticks repo to promise, lets keep the benefits to tehnical ones - as this is builders technical documentation

**Comment by gluneau** [comment]
Updates to comments upstream
https://github.com/AcalaNetwork/chopsticks/pull/296

File: docs/build/environment/chopsticks.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+6. Streamlined consensus-building: Forking a live blockchain enables developers to present real-world test results to stakeholders, which can help build consensus for proposed changes. This can expedite the decision-making process and lead to a more efficient and effective governance model.
+
+7. Simplified debugging: Debugging transactions and smart contracts on a forked version of the blockchain allows developers to isolate and address issues more easily, ensuring that only well-vetted code is introduced to the main network.
+
+8. Enhanced scalability: By testing proposed changes on a forked version of the live blockchain, developers can identify scalability bottlenecks and implement solutions to support future growth, ultimately contributing to the long-term success of the blockchain network.
```
[the remark this answers is not in the document]
Comment (gluneau) [reply docs/build/environment/chopsticks.md]: #8 removed

File: docs/build/environment/chopsticks.md (hunk 1 above)
Comment (gluneau) [reply docs/build/environment/chopsticks.md]: I've removed the optimistic sentiment sentences.

**Comment by impelcrypto** [comment]
It seems that the staging URL is not working on my end. @niklabh Do you know what might be the issue?
https://astar-docs--pr337-chopsticks-anp1w9k4.web.app/

image

**Review by impelcrypto — APPROVED** [review]

**Review by sirius651 — APPROVED** [review]

**Review by fiexer — APPROVED** [review]

**Pull request merged**